src/azure-cli/azure/cli/command_modules/synapse/_params.py

```diff
@@ -251,29 +251,44 @@ def load_arguments(self, _):
     with self.argument_context('synapse sql pool audit-policy') as c:
         c.argument('sql_pool_name', arg_type=name_type, id_part='child_name_1', help='The SQL pool name.')
 
-    with self.argument_context('synapse sql pool audit-policy update') as c:
-        _configure_security_or_audit_policy_storage_params(c)
-        c.argument('storage_account_subscription_id', arg_group=storage_arg_group,
-                   options_list=['--storage-subscription'],
-                   help='The subscription id of storage account')
-        c.argument('is_storage_secondary_key_in_use', arg_group=storage_arg_group,
-                   arg_type=get_three_state_flag(), options_list=['--use-secondary-key'],
-                   help='Indicates whether using the secondary storeage key or not')
-        c.argument('is_azure_monitor_target_enabled', options_list=['--enable-azure-monitor'],
-                   help='Whether enabling azure monitor target or not.',
-                   arg_type=get_three_state_flag())
-        c.argument('state',
-                   arg_group=policy_arg_group,
-                   help='Auditing policy state',
-                   arg_type=get_enum_type(BlobAuditingPolicyState))
-        c.argument('audit_actions_and_groups',
-                   options_list=['--actions'],
-                   arg_group=policy_arg_group,
-                   help='List of actions and action groups to audit.',
-                   nargs='+')
-        c.argument('retention_days',
-                   arg_group=policy_arg_group,
-                   help='The number of days to retain audit logs.')
+    for scope in ['synapse sql pool audit-policy', 'synapse sql audit-policy']:
+        with self.argument_context(scope + ' update') as c:
+            _configure_security_or_audit_policy_storage_params(c)
+            c.argument('storage_account_subscription_id', arg_group=storage_arg_group,
```

Comment:
actually I am curious about previous design. If you want to  use storage account in another subscription, why not declare a storage account parameter which would accept both storage account resource id and name? 
In this way, cx could use resource id to access storage account in another subscription and use name for storage account in the same subscription.

Reply:
> actually I am curious about previous design. If you want to use storage account in another subscription, why not declare a storage account parameter which would accept both storage account resource id and name?
> In this way, cx could use resource id to access storage account in another subscription and use name for storage account in the same subscription.

I agree with you. In fact storage account name is global unique. But in the api definition (please see this [line](https://github.com/Azure/azure-rest-api-specs/blob/6bf460bee8e1d3a837fd0e37b1864cba4725c124/specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json#L5481)), there is a property requires subscription id. Per my previous experience, I usually define a related parameter for property in the api. As you see the update cmdlet can update a property.

Reply:
I see.